app/code/Magento/SalesSequence/Observer/StoreDeleteObserver.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\SalesSequence\Observer;
+
+use Magento\Framework\Event\Observer as EventObserver;
+use Magento\SalesSequence\Model\Builder;
+use Magento\SalesSequence\Model\EntityPool;
+use Magento\Framework\Event\ObserverInterface;
+
+/**
+ * Class StoreDeleteObserver
+ */
+class StoreDeleteObserver implements ObserverInterface
+{
+    /**
+     * @var Builder
+     */
+    private $sequenceBuilder;
+
+    /**
+     * @var EntityPool
+     */
+    private $entityPool;
+
+    /**
+     * Initialization
+     *
+     * @param Builder $sequenceBuilder
+     * @param EntityPool $entityPool
+     */
+    public function __construct(
+        Builder $sequenceBuilder,
+        EntityPool $entityPool
+    ) {
+        $this->sequenceBuilder = $sequenceBuilder;
+        $this->entityPool = $entityPool;
+    }
+
+    /**
+     * @param EventObserver $observer
+     * @return $this
+     */
+    public function execute(EventObserver $observer)
+    {
+        $storeId = $observer->getData('store')->getId();
+        foreach ($this->entityPool->getEntities() as $entityType) {
```

Comment:
I think it will be better from a performance point of view to load all meta_ids using a single query, also entityType is not really required for this operation, as the delete operation is performed for all the entity types